# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/wifi/nsWifiMonitor.h
# Commit: 35c408b9531d
# Full Hash: 35c408b9531db01c71dee288cab0a02a633fb535
# Author: David Parks <daparks@mozilla.com>
# Date: 2023-05-13 21:42:25
# Regressor Bug: 1810421
# File Overlap Count: 2
# Description:
#   Bug 1810421: Refactor wifi scanning to consolidate generic code r=necko-reviewers,kershaw,valentin
#   
#   Reorganizes the wifi-scanning code and makes more of it platform-generic
#   to ease the transition from polling the wifi to usually scanning only on
#   network changes.  This is essentially just moving files/code around and
# ==============================================================================

diff -r 029be11fa3d8 -r 35c408b9531d netwerk/wifi/nsWifiMonitor.h
--- a/netwerk/wifi/nsWifiMonitor.h	Sat May 13 00:11:51 2023 +0000
+++ b/netwerk/wifi/nsWifiMonitor.h	Sat May 13 00:36:00 2023 +0000
@@ -18,10 +18,7 @@
 #include "nsIObserver.h"
 #include "nsTArray.h"
 #include "mozilla/Attributes.h"
-
-#ifdef XP_WIN
-#  include "win_wifiScanner.h"
-#endif
+#include "WifiScanner.h"
 
 extern mozilla::LazyLogModule gWifiMonitorLog;
 #define LOG(args) MOZ_LOG(gWifiMonitorLog, mozilla::LogLevel::Debug, args)
@@ -55,7 +52,8 @@
   NS_DECL_NSIRUNNABLE
   NS_DECL_NSIOBSERVER
 
-  nsWifiMonitor();
+  explicit nsWifiMonitor(
+      mozilla::UniquePtr<mozilla::WifiScanner>&& aScanner = nullptr);
 
  private:
   ~nsWifiMonitor() = default;
@@ -71,13 +69,12 @@
   mozilla::Atomic<bool> mThreadComplete;
   nsCOMPtr<nsIThread> mThread;  // only accessed on MainThread
 
+  // Background thread only (except in test).
+  mozilla::UniquePtr<mozilla::WifiScanner> mWifiScanner;
+
   nsTArray<nsWifiListener> mListeners MOZ_GUARDED_BY(mReentrantMonitor);
 
   mozilla::ReentrantMonitor mReentrantMonitor;
-
-#ifdef XP_WIN
-  mozilla::UniquePtr<WinWifiScanner> mWinWifiScanner;
-#endif
 };
 
 #endif